src/App.js

```diff
@@ -1,16 +1,96 @@
-import React from 'react';
+import React from "react";
+import Search from "./Search";
+import Results from "./Results";
+// import Info from "./Info";
+
+let movieArray = [];
 
 class App extends React.Component {
-  constructor(){
+  constructor(props) {
     super();
+    this.state = {
+      movies: [],
+      currentMovie: "",
+      currentInfo: "",
+      sortBy: ""
+
+    };
+    this.sortMovie = this.sortMovie.bind(this)
+    this.receiver = this.receiver.bind(this);
+    this.receiveMovie = this.receiveMovie.bind(this);
+    this.clearResults = this.clearResults.bind(this);
+  }
+
+  //receiver function to populate movies in state
+  receiver(results) {
+    movieArray.unshift(results);
+    this.setState({ movies: movieArray });
+    console.log(this.state.movies)
+  }
+
+  //clear current results so that more movies can be fetched
+  clearResults() {
+    movieArray.length = 0;
+  }
+
+  //receive user-clicked movie to set more info state variables
+  receiveMovie(movieID, plot) {
+    this.setState({
+      ["currentMovie"]: movieID,
+      ["currentInfo"]: plot
+    });
   }
 
-  render(){
+  //sort movies by imdbRating or Year depending on which radio button was pushed
+  sortMovie(event) {
+    event.preventDefault();
+    this.setState({
+      sortBy: event.target.name
+    })
+
+    const sorted = (event.target.name === "Rating") ? this.state.movies.sort((a, b) => {
```

Comment:
This call will mutate state. It would be better to create a clone of `this.state.movies`, sort it, then set it back in state using `setState`.